build/kconfig/src/lib.rs

```diff
@@ -0,0 +1,145 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use serde::{Deserialize, Serialize};
+use std::collections::{BTreeMap, BTreeSet};
+
+/// Application configuration passed into the kernel build.
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct KernelConfig {
+    /// Tasks in the app image. The order of tasks is significant.
+    pub tasks: Vec<TaskConfig>,
+
+    /// Regions that tasks have shared access to, keyed by the name the task
+    /// config used to grant access (often peripheral name). These are typically
+    /// memory mapped peripherals.
+    pub shared_regions: BTreeMap<String, RegionConfig>,
+
+    /// Interrupts hooked by the application, keyed by IRQ number.
+    pub irqs: BTreeMap<u32, InterruptConfig>,
+
+    pub timer: (u32, u32),
```

Comment:
Missing comment. 

Also, this seems to be a Rivos-specific change. Probably add a note to commit message outlining what's changed in this commit compared to what's upstream?

Reply:
Good idea, will do.

Reply:
Thanks Fawaz. We might want to adopt some process to make it easier for the reviewers to understand which changes are picked as is v/s changes that were adapted. Reference: https://docs.google.com/document/d/1XcFBzz9hLu5ryTEKyMMIO3CcbY-BG-c3otyo4H1uD3E/edit. We can probably use tags like "BACKPORT" and "UPSTREAM" which should make it pretty clear. Anyways, something for the future.